src/gt4py/backend/gtc_backend/cuda/backend.py

```diff
@@ -19,43 +19,44 @@
 from eve import codegen
 from gt4py import gt_src_manager
 from gt4py.backend.base import CLIBackendMixin, register
-from gt4py.backend.gt_backends import (
+from gt4py.backend.gtc_backend.base import GeneratorClass
+from gt4py.backend.gtc_backend.common import bindings_main_template, pybuffer_to_sid
+from gtc import gtir
+from gtc.common import DataType
+from gtc.cuir import cuir, cuir_codegen, extent_analysis, kernel_fusion
+from gtc.cuir.oir_to_cuir import OIRToCUIR
+from gtc.gtir_to_oir import GTIRToOIR
+from gtc.passes.gtir_pipeline import GtirPipeline
+from gtc.passes.oir_optimizations.pruning import NoFieldAccessPruning
+from gtc.passes.oir_pipeline import DefaultPipeline
+
+from ..base import (
     BaseGTBackend,
     GTCUDAPyModuleGenerator,
     cuda_is_compatible_layout,
     cuda_is_compatible_type,
     make_cuda_layout_map,
 )
-from gt4py.backend.gtc_backend.common import bindings_main_template, pybuffer_to_sid
-from gt4py.backend.gtc_backend.defir_to_gtir import DefIRToGTIR
-from gtc import gtir_to_oir
-from gtc.common import DataType
-from gtc.cuir import cuir, cuir_codegen, extent_analysis, kernel_fusion, oir_to_cuir
-from gtc.passes.gtir_pipeline import GtirPipeline
-from gtc.passes.oir_optimizations.caches import FillFlushToLocalKCaches
-from gtc.passes.oir_optimizations.pruning import NoFieldAccessPruning
-from gtc.passes.oir_pipeline import DefaultPipeline
 
 
 if TYPE_CHECKING:
     from gt4py.stencil_object import StencilObject
 
 
-class GTCCudaExtGenerator:
+class GTCCudaExtGenerator(GeneratorClass):
     def __init__(self, class_name, module_name, backend):
         self.class_name = class_name
         self.module_name = module_name
         self.backend = backend
 
-    def __call__(self, definition_ir) -> Dict[str, Dict[str, str]]:
-        gtir = GtirPipeline(DefIRToGTIR.apply(definition_ir)).full()
-        base_oir = gtir_to_oir.GTIRToOIR().visit(gtir)
+    def __call__(self, ir: gtir.Stencil) -> Dict[str, Dict[str, str]]:
```

Comment:
Find a better naming scheme for the `gtir` here and elsewhere.

Reply:
the name `gtir` was shadowing the imported module in a few spots. Since it's the "default" IR now I thought perhaps `ir` was best.